### PR TITLE
[16.0][ADD] lighting_reporting_product

### PR DESCRIPTION
**Pull request by FrankC013**

### CODE REVIEW

File: lighting_reporting_product/controllers/main.py [hunk 1]
```diff
@@ -54,17 +60,35 @@ def extract_params(self, param1, param2):
...
+            )
+        return company
+
     @http.route(
         [
             "/web/datasheet/<string:reference>",
             "/web/datasheet/<string:param1>/<string:reference>",
-            "/web/datasheet/<string:param2>/<string:reference>",
             "/web/datasheet/<string:param1>/<string:param2>/<string:reference>",
+            "/web/datasheet/<string:company_id>/<string:param1>/<string:param2>"
```
Comment (eantones) [inline lighting_reporting_product/controllers/main.py]:
```suggestion
            "/web/datasheet/<int:company_id>/<string:param1>/<string:param2>"
```

File: lighting_reporting_vanguard_product/report/report_vanguard.py [hunk 2]
```diff
@@ -12,6 +12,7 @@ class ProductReport(models.AbstractModel):
     def _get_report_values(self, docids, data=None):
         model = "lighting.product"
         lang = self.env.lang
+        company = self.env.company
```
Comment (eantones) [inline lighting_reporting_vanguard_product/report/report_vanguard.py:15]: This and all the other changes should be done on Classic as well. During test phase we're gonna need Classic to work in production and it cannot fail at all. No only Vanguard but Classic should also be tested intensively.

**Review by eantones — CHANGES_REQUESTED** [review]

**Comment by FrankC013** [comment]
@eantones Pending to merge #53 and rebase

File: lighting_reporting_vanguard_product/views/report_vanguard.xml [hunk 3]
```diff
@@ -47,6 +47,7 @@
                             </t>
                             <t t-if="len(doc.catalog_ids) == 1">
                                 <img
+                                    t-if="doc.catalog_ids[0].image_512"
                                     t-att-src="image_data_uri(doc.catalog_ids[0].image_512)"
```
Comment (eantones) [inline lighting_reporting_vanguard_product/views/report_vanguard.xml]: Put "doc.catalog_ids[0].image_512" in a t-value/t-set

File: lighting_reporting_vanguard_product/views/report_vanguard.xml [hunk 4]
```diff
@@ -307,43 +308,31 @@
                                                     <td />
                                                 </tr>
                                                 <tr
-                                                    t-if="line.color_temperature_flux_ids.filtered(lambda x: x.nominal_flux)"
+                                                    t-if="line.get_dependent_field_value('color_temperature_flux_ids')"
```
Comment (eantones) [inline lighting_reporting_vanguard_product/views/report_vanguard.xml:315]: Remember, it's not the same
```suggestion
                                                    t-if="line.get_dependent_field_value('color_temperature_flux_ids')" is not None"
```
do the same on the other ones

**Review by eantones — CHANGES_REQUESTED** [review]

**Review by eantones — APPROVED** [review]

**Comment by FrankC013** [comment]
Pending to merge #53 and rebase

**Review by eantones — CHANGES_REQUESTED** [review]
Please squash

**Comment by FrankC013** [comment]
@eantones I had to use a combined approach with Python because if I only use the template tools, the decimal separator is not respected

File: lighting/models/product_source_line.py [hunk 5]
```diff
@@ -428,6 +429,43 @@ def _compute_color_temperature_flux_display(self):
...
+            flux_values = []
+            for flux in rec.color_temperature_flux_ids:
+                flux_values.append(
+                    "%s%s"
+                    % (
+                        self.get_format_lang_decimal(flux.total_flux),
+                        flux.flux_magnitude,
+                    )
+                )
+            rec.total_flux_display = ", ".join(flux_values)
```
Comment (eantones) [inline lighting/models/product_source_line.py]: These codes are identical.

**Review by eantones — CHANGES_REQUESTED** [review]

**Review by eantones — APPROVED** [review]

File: lighting/models/product_source_line.py [hunk 6]
```diff
@@ -428,6 +429,47 @@ def _compute_color_temperature_flux_display(self):
...
+            return False
+        return "/".join(
+            "%s"
+            % (
+                self.get_format_lang_decimal(flux_value) + flux_magnitude
+                if flux_value
+                else "-"
+            )
+            for flux_value, flux_magnitude in flux_data
+        )
```
Comment (eantones) [inline lighting/models/product_source_line.py]:
```suggestion
def _get_color_temperature_flux_values(self, flux_attr):
    self.ensure_one()
    found = False
    flux_data = []
    for flux in self.color_temperature_flux_ids:
        value = flux[flux_attr]
        if value:
            value = self.get_format_lang_decimal(value) + flux.flux_magnitude
            if not found:
                found = True
        else:
            value = "-"
        flux_data.append(value)
    return found and "/".join(flux_data)
```

**Review by eantones — CHANGES_REQUESTED** [review]

File: lighting/models/product_source_line.py [hunk 7]
```diff
@@ -428,6 +429,46 @@ def _compute_color_temperature_flux_display(self):
...
+            if value:
+                value = self.get_format_lang_decimal(value) + flux.flux_magnitude
+                if not found:
+                    found = True
+            else:
+                value = "-"
+            flux_data.append(value)
+        if found:
+            return "/".join(flux_data)
+        return found
```
Comment (eantones) [inline lighting/models/product_source_line.py]:
```suggestion
        return found and "/".join(flux_data)
```

**Review by eantones — CHANGES_REQUESTED** [review]

**Review by eantones — APPROVED** [review]

**Pull request merged**